File: Programming/JAVA_programming/JAVA_CORE practise programs/dyanamicmemory.java
import java.util.Scanner;

class dyanamicmemory
{
    public static void main(String A[])
    {
        Scanner sobj=new Scanner(System.in);
        int size=0;
        
        System.out.println("enter number of elements:");
        size = sobj.nextInt();

        //dynyamic memory allocation
        float marks[]=new float[size];
        
        //use the memory
        for(int i=0; i<size; i++)
        {
            System.out.println("enter marks for student "+(i+1)+":");
            marks[i]=sobj.nextFloat();
        }
        
        System.out.println("\nMarks entered:");
        for(int i=0; i<size; i++)
        {
            System.out.println("Student "+(i+1)+": "+marks[i]);
        }
        
        marks=null;
        System.gc(); //garbage collector
        sobj.close();

    }






}

File: Programming/C++_programming/C++_CORE practice/dyanamicmemory.cpp
#include<iostream>
using namespace std;

int main()
{

    int size =0;
    float *marks = NULL;
    int i=0;

    cout<<"enter your number of elements:  \n ";
    cin>>size;

    //dyanamic memory allocation
    marks=new float[size];


    cout<<("enter your marks:  \n");

    //iteration
    for(i=0; i<size; i++)
    {
        cin>>  marks[i];

    }
    cout<<"entered marks are:  \n";

    for ( i = 0; i < size; i++)
    {
        cout<<marks[i]<<"\n";

    }
    
delete[]marks;

return  0;







}